pkg/artifactory/datasource_artifactory_file.go

```diff
@@ -114,50 +138,66 @@ func dataSourceFileReader(ctx context.Context, d *schema.ResourceData, m interfa
 	path := d.Get("path").(string)
 	outputPath := d.Get("output_path").(string)
 	forceOverwrite := d.Get("force_overwrite").(bool)
+	downloadLatest := d.Get("download_latest_artifact").(bool)
 	fileInfo := FileInfo{}
-	_, err := m.(*resty.Client).R().SetResult(&fileInfo).Get(fmt.Sprintf("artifactory/api/storage/%s/%s", repository, path))
-	if err != nil {
-		return diag.FromErr(err)
-	}
 
-	fileExists := FileExists(outputPath)
-	chksMatches, _ := VerifySha256Checksum(outputPath, fileInfo.Checksums.Sha256)
-
-	/*--File Download logic--
-	1. File doesn't exist
-	2. In Data Source argument `force_overwrite` set to true, an existing file in the output_path will be overwritten. Ignore file exists or not
-	3. File exists but check sum doesn't match
-	*/
-	if !fileExists || forceOverwrite || (fileExists && !chksMatches) {
-		outdir := filepath.Dir(outputPath)
-		err = os.MkdirAll(outdir, os.ModePerm)
+	if downloadLatest == false {
+		response, err := m.(*resty.Client).R().SetResult(&fileInfo).Get(fmt.Sprintf("artifactory/api/storage/%s/%s", repository, path))
+		path := (fmt.Sprintf("artifactory/api/storage/%s/%s", repository, path))
+		fmt.Printf(path)
+		fmt.Print(response.StatusCode())
 		if err != nil {
 			return diag.FromErr(err)
 		}
-		outFile, err := os.Create(outputPath)
+
```

Comment:
You've messed up your PR - this code is already in master but is showing as a diff here. Better rebase